metadata-io/src/main/java/com/linkedin/metadata/schema/registry/SchemaRegistryServiceImpl.java

```diff
@@ -0,0 +1,64 @@
+package com.linkedin.metadata.schema.registry;
+
+import com.google.common.collect.BiMap;
+import com.google.common.collect.HashBiMap;
+import com.linkedin.mxe.TopicConvention;
+import com.linkedin.pegasus2avro.mxe.FailedMetadataChangeEvent;
+import com.linkedin.pegasus2avro.mxe.FailedMetadataChangeProposal;
+import com.linkedin.pegasus2avro.mxe.MetadataAuditEvent;
+import com.linkedin.pegasus2avro.mxe.MetadataChangeEvent;
+import com.linkedin.pegasus2avro.mxe.MetadataChangeLog;
+import com.linkedin.pegasus2avro.mxe.MetadataChangeProposal;
+import com.linkedin.pegasus2avro.mxe.PlatformEvent;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import org.apache.avro.Schema;
+
+
+public class SchemaRegistryServiceImpl implements SchemaRegistryService {
+
+  private final Map<String, Schema> _schemaMap;
+
+  private final BiMap<String, Integer> _subjectToIdMap;
+
+  public SchemaRegistryServiceImpl(final TopicConvention convention) {
+    this._schemaMap = new HashMap<>();
+    this._subjectToIdMap = HashBiMap.create();
+    this._schemaMap.put(convention.getMetadataChangeProposalTopicName(), MetadataChangeProposal.getClassSchema());
+    this._subjectToIdMap.put(convention.getMetadataChangeProposalTopicName(), 1);
```

Comment:
Can we make these integers into an Enum and use the ordinal?

Reply:
I just mean something like:

```Java
enum TOPIC_ORDINAL {
MCP_TOPIC
MCL_TOPIC
...
```

and then:

```Java
this._schemaMap.put(convention.getMetadataChangeProposalTopicName(), MCP_TOPIC.ordinal());
```

Reply:
That would mean that the order in the enum would have to match the order of the `schemaMap.put` operations.

```
enum TOPIC_ORDINAL {
MCP_TOPIC
MCL_TOPIC
....

public SchemaRegistryServiceImpl(final TopicConvention convention) {
    this._schemaMap = new HashMap<>();
    this._subjectToIdMap = HashBiMap.create();
    this._schemaMap.put(convention.getMetadataChangeProposalTopicName(), MetadataChangeProposal.getClassSchema());
    this._subjectToIdMap.put(convention.getMetadataChangeProposalTopicName(), MCP_TOPIC.ordinal());
    this._schemaMap.put(convention.getMetadataChangeLogVersionedTopicName(), MetadataChangeLog.getClassSchema());
    this._subjectToIdMap.put(convention.getMetadataChangeLogVersionedTopicName(), MCL_TOPIC.ordinal());
```

Reply:
They're already ordered anyway though, any modifications to the list of topics has to account for that already and this makes it more conducive to programmatic access of the id -> topic name direction of the BiMap.

Reply:
Please take another look

Reply:
LGTM!